src/window.cpp

```diff
@@ -805,7 +805,10 @@ static void DispatchRightClickEvent(Window *w, int x, int y)
 	}
 
 	/* Right-click close is enabled and there is a closebox */
-	if (_settings_client.gui.right_mouse_wnd_close && w->nested_root->GetWidgetOfType(WWT_CLOSEBOX)) {
+	if (_settings_client.gui.right_mouse_wnd_close == CLOSE && w->nested_root->GetWidgetOfType(WWT_CLOSEBOX)) {
+		w->Close();
+	/* Right-click close is enabled, but excluding pinned windows */
```

Comment:
Indentation

Reply:
Comment matches the `else if` below, unless you mean something else?

Reply:
The comment is in the wrong place. It is unclear it is a comment for the `else if` statement. Instead, if you move it inside of the body of that statement, and indent it, it becomes much more clear what is going on. So something like:

```
} else if (...) {
    /* Right-click close is enabled, but excluding pinned windows. */
   w->Close();
```

PS: missing dot at end of sentence in that comment. Please add that. Years ago we changed that sentences should end in dots; it is only older code that doesn't do that yet.